crates/xtask/src/modify_data.rs

```diff
@@ -0,0 +1,225 @@
+
+        #[cfg(not(windows))]
+        {
+            let uid = users::get_current_uid();
+            let gid = users::get_current_gid();
+
+            docker_image = docker_image
+                .with_env_var(("PLEX_UID", uid.to_string()))
+                .with_env_var(("PLEX_GID", gid.to_string()));
+        }
+
+        let docker = clients::Cli::default();
+
+        print!("// Spawning docker container... ");
+        let _ = std::io::stdout().flush();
+
+        let plex_node = docker.run(docker_image);
+
+        println!("done!");
+        let _ = std::io::stdout().flush();
+
+        let server_url = format!("http://localhost:{}/", plex_node.get_host_port_ipv4(32400));
+        cmd!("cargo run -p plex-cli --  --server {server_url} wait").run()?;
+
+        print!(
+            "// Server is running at {}web/index.html. Press enter when done...",
+            server_url
+        );
+        let _ = std::io::stdout().flush();
+
+        let mut input_string = String::new();
+        std::io::stdin().read_line(&mut input_string).unwrap();
+
+        print!("// Stopping docker container... ");
+        let _ = std::io::stdout().flush();
+        plex_node.stop();
+        println!("done!");
+
+        self.copy_tree(&plex_config_path, &plex_stub_config_path, vec!["Media"])?;
+        self.copy_tree(&plex_config_path, &plex_stub_config_path, vec!["Metadata"])?;
+        self.copy_tree(
+            &plex_config_path,
+            &plex_stub_config_path,
+            vec!["Plug-in Support", "Databases"],
+        )?;
+
+        Ok(())
+    }
+
+    fn copy_tree(
+        &self,
+        plex_config_path: &Path,
+        plex_stub_config_path: &Path,
+        paths: Vec<&str>,
+    ) -> anyhow::Result<()> {
+        let mut source = plex_config_path.to_owned();
+        let mut target = plex_stub_config_path.to_owned();
+
+        for path in paths {
+            source = source.join(path);
+            target = target.join(path);
+        }
+
+        remove_dir_all(&target)?;
+
+        for entry in WalkDir::new(&source)
+            .follow_links(false)
+            .into_iter()
+            .filter_map(|e| e.ok())
+        {
+            if entry.file_type().is_file() {
+                let path = entry.path();
+                let new_path = target.join(path.strip_prefix(&source)?);
+
+                self.cp(entry.path(), new_path)?;
+            } else if entry.file_type().is_symlink() {
+                let path = entry.path();
+                let new_path = plex_stub_config_path.join(path.strip_prefix(plex_config_path)?);
+                let link_target = path.canonicalize()?;
+
+                if link_target.strip_prefix(plex_config_path).is_ok() {
+                    self.symlink(path, &link_target, &new_path)?;
+                } else {
+                    self.cp(path, new_path)?;
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    fn symlink<P: AsRef<Path>, Q: AsRef<Path>, R: AsRef<Path>>(
```

Comment:
Would you mind moving this method elsewhere since it's defined inside two different commands now?